scripts/python/packages/parthenon_tools/parthenon_tools/movie2d.py

```diff
@@ -142,45 +172,75 @@ def plot_dump(
+
+    if args.tc and args.vc:
         raise ValueError(
             "Only one of --tensor-component and --vector-component should be set."
         )
+
+    if args.workers > 1:
+        logger.warning(
+            "Matplotlib is not multi-thread friendly. Read this for more details https://matplotlib.org/stable/users/faq/howto_faq.html#work-with-threads"
+        )
+        logger.warning(
+            "Try decreasing threads if you encounter any undefined behaviour"
+        )
+    # Create output director if does't exists
+    args.output_directory.mkdir(0o755, True, True)
+    logger.info(f"Total files to process: {len(args.files)}")
+
+    components = [0, 0]
     if args.tc is not None:
         components = args.tc
     if args.vc is not None:
         components = [0, args.vc]
-    dump_id = 0
-    debug_plot = False
-    for f in files:
-        data = read(f)
-        print(data)
-        q = data.Get(field, False, not debug_plot)
-        name = str(dump_id).rjust(4, "0") + ".png"
-        if debug_plot:
-            plot_dump(
-                data.xg,
-                data.yg,
-                q,
-                name,
-                True,
-                data.gid,
-                data.xig,
-                data.yig,
-                data.xeg,
-                data.yeg,
-                components,
-            )
-        else:
-            plot_dump(data.xng, data.yng, q, name, True)
-        dump_id += 1
+
+    _x = ProcessPoolExecutor if args.worker_type == "process" else ThreadPoolExecutor
+    current_time = 0.0 if args.time_step else None
+    with _x(max_workers=args.workers) as pool:
+        for dump_id, file_name in enumerate(args.files):
+            data = phdf(file_name)
+            if args.field not in data.Variables:
+                logging.error(
+                    f'No such field "{args.field}" in {file_name}. \
+This will lead to stop further processing'
+                )
+                break
+
+            logger.debug(f"Submitting {file_name}")
+            q = data.Get(args.field, False, not args.debug)
+            name = args.output_directory / Path(str(dump_id).rjust(4, "0") + ".png")
+            if args.debug:
+                pool.submit(
+                    plot_dump,
+                    data.xg,
+                    data.yg,
+                    q,
+                    current_time,
+                    name,
+                    True,
+                    data.gid,
+                    data.xig,
+                    data.yig,
+                    data.xeg,
+                    data.yeg,
+                    components,
+                )
+            else:
+                pool.submit(plot_dump, data.xng, data.yng, q, current_time, name, True)
+            if args.time_step:
+                current_time += args.time_step
+                current_time = round(current_time, ndigits=2)
+
+    logger.info("All files are sent to the processor")
```

Comment:
@Yurlungur I couldn't get what you are saying. Could you give more information here?

Reply:
Ohk, but how can we check for the field and break the loop

```py
if args.field not in data.Variables:
                logging.error(
                    f'No such field "{args.field}" in {file_name}. \
This will lead to stop further processing'
                )
                break
```

Reply:
Your version produces a cleaner error message, so that's fine. It's just a little bit more concise to construct an array of args to pass in and then `pool.map(make_frame_worker, args)`. You'll still get a useful error message out, but it'll be produced asynchronously by the worker that failed.

Although actually I just thought of something: Currently the `print(data)` line means `movie2d` outputs all the available variables so if I for example call
```python
movie2d.py nonexistent_var my_run.out0.00000.phdf
```
it will output the variables available to plot before the code dies. Your version does not do this. Any error message that runs the movie2d script should provide this so that users can easily get some introspection about the mesh.

Reply:
ACK

Reply:
I am now showing all the variables

![Screenshot_20230405_180900](https://user-images.githubusercontent.com/28386721/230083072-d9017176-afb6-4ae9-8d06-4b9b5c7ed656.png)